air/src/air/transition/mod.rs

```diff
@@ -31,7 +31,7 @@ pub struct TransitionConstraints<E: FieldElement> {
     main_constraint_degrees: Vec<TransitionConstraintDegree>,
     aux_constraints: Vec<TransitionConstraintGroup<E>>,
     aux_constraint_degrees: Vec<TransitionConstraintDegree>,
-    divisor: ConstraintDivisor<E::BaseField>,
+    divisors: Vec<ConstraintDivisor<E::BaseField>>,
```

Comment:
You're setting the number of divisors as the number of degrees? (and I think in the end all the divisors would be the same). I was thinking more about a `main_divisor` and `aux_divisor` or at least, many constraints should share the same divisor.  Otherwise it might be heavy to the verifier (don't know how much but I'm worried because is linear in the number of divisors)

Reply:
Yes, but they are grouped together according to divisor degree when evaluating, so there isn't unnecessary work being done. You're right that in most cases constraints should share the same divisor, but I have a need to specify different divisors in a small fraction of constraints.

Reply:
And I'm mistaken again -- it is possible to prove execution with just single `main_divisor` and `aux_divisor` for the Cairo constraints. I'm still wondering if there are use cases that call for a more granular control of the divisors. As I mentioned above I don't think my implementation is adding too much unnecessary work, but an argument could be made for simplifying things if there aren't any use cases for this.